org.eclipse.sisu.inject/pom.xml

```diff
@@ -395,4 +395,46 @@
     </plugins>
   </build>
 
+  <profiles>
+    <profile>
+      <id>sonatype-oss-release</id>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-shade-plugin</artifactId>
+            <executions>
+              <execution>
+                <phase>package</phase>
+                <goals>
+                  <goal>shade</goal>
+                </goals>
+                <configuration>
+                  <!--
+                   | Create 'no_asm' jar which doesn't embed ASM and refers to the original org.objectweb.asm package.
+                  -->
+                  <shadedArtifactAttached>true</shadedArtifactAttached>
+                  <shadedClassifierName>no_asm</shadedClassifierName>
+                  <filters>
+                    <filter>
+                      <artifact>org.eclipse.sisu:org.eclipse.sisu.inject</artifact>
+                      <excludes>
+                        <exclude>org/eclipse/sisu/space/asm/**</exclude>
+                      </excludes>
+                    </filter>
+                  </filters>
+                  <relocations>
+                    <relocation>
+                      <pattern>org.eclipse.sisu.space.asm</pattern>
```

Comment:
This also requires additional import-package headers for OSGi

Reply:
I took the easy route and added a `DynamicImport-Package` entry to the "no_asm" jar, rather than mess around generating multiple manifests. Most consumers of the "no_asm" jar won't be deploying it on OSGi, but if any do then it will work now.